### ISSUE TITLE
protect text inside of anchor link keys

### ISSUE THREAD

**Issue by zkamvar** [ropensci/tinkr#85]
Whilst planning the transition for Carpentries lessons, I ran into a situation where a page with anchor links were being mangled: 

``` r
txt <- "**Maintainer(s):**


* [Sarah Brown][brown_sarah]: @brownsarahm
* [Tim Dennis][dennis_tim]: @jt14den
* [David Perez-Suarez][perez-suarez_david]: @dpshelio
* [Nathaniel Porter][porter-nathaniel]: @ndporter
* [Jon Wheeler][wheeler_jon]: @jonathanwheeler01
* [Karen Word][word_karen]: @karenword  


[dc-site]: http://datacarpentry.org
[lesson-example]: https://carpentries.github.io/lesson-example
[swc-site]: http://software-carpentry.org
[lc-site]: https://librarycarpentry.org
[koch_christina]: https://carpentries.org/instructors/
[brown_sarah]: https://carpentries.org/instructors/ 
[dennis_tim]: https://carpentries.org/instructors/
[perez-suarez_david]: https://carpentries.org/instructors/
[porter-nathaniel]: https://carpentries.org/instructors/
[wheeler_jon]: https://carpentries.org/instructors/
[word_karen]: https://carpentries.org/team/"
tmp <- tempfile()
writeLines(txt, tmp)
tinkr::yarn$new(tmp)$show()
#> **Maintainer(s):**
#> 
#> - [Sarah Brown][wheeler_jon]: @brownsarahm
#> - [Tim Dennis][wheeler_jon]: @jt14den
#> - [David Perez-Suarez][wheeler_jon]: @dpshelio
#> - [Nathaniel Porter][wheeler_jon]: @ndporter
#> - [Jon Wheeler][wheeler_jon]: @jonathanwheeler01
#> - [Karen Word][word_karen]: @karenword
#> 
#> [wheeler\_jon]: https://carpentries.org/instructors/
#> [word\_karen]: https://carpentries.org/team/
```

<sup>Created on 2023-01-25 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

There are actually two issues here:

1. anchor links with duplicate links are coalesced into the last one
2. the underscore character is not protected in the anchor link definition

**Comment by zkamvar** [comment]
I seem to have forgotten to include relevant context:

https://github.com/carpentries/lesson-transition/issues/15

**Comment by zkamvar** [comment]
The fix for escaping the anchor links can be achieved by modifying the anchor link template if we modify line 48 to be `<xsl:value-of select='string(.)'/>`:

https://github.com/ropensci/tinkr/blob/e99436a81796442c84d7192e89a0b734c4ffff84/inst/stylesheets/xml2md_gfm.xsl#L45-L63

**Comment by zkamvar** [comment]
I believe this was fixed in #86
